lib/formatters/tapFormatter.mjs

```diff
@@ -1,15 +1,16 @@
 import { isNumber } from '../utils/validateTypes.mjs';
 import preprocessWarnings from './preprocessWarnings.mjs';
 
-/** @import {Warning} from 'stylelint' */
+/** @import {Warning, Formatter, LintResult} from 'stylelint' */
 /** @typedef {Array<Omit<Warning, 'rule'>>} Warnings */
 
-/** @type {import('stylelint').Formatter} */
+/** @type {Formatter} */
 export default function tapFormatter(results) {
 	const lines = ['TAP version 14', `1..${results.length}`];
 
 	for (const [index, result] of results.entries()) {
 		preprocessWarnings(result);
+		escapeOctothorp(result);
```

Comment:
I don't think it is correct to mutate the `result`.
We only want to escape the output from this formatter we don't want to modify the `result.source` in ways that could be observable.

Reply:
`preprocessWarnings` does it too.
i.e. mutating result

I could return a copy but that would require to refactor `preprocessWarnings` and I didn't want to do that in this PR.

Reply:
We can avoid extra mutations without refactoring `preprocessWarnings`.

Reply:
Either result is not mutated at all or it can be.
It's an opinion.
i.e. I value consistency

Let's discuss this again, after Iv done the changes that you requested in https://github.com/stylelint/stylelint/issues/8321#issuecomment-2613924041.